### ISSUE TITLE
Issue with module names containing `.`

### ISSUE THREAD

**Issue by jacobbuck** [jrburke/r.js#789]
I'm running into an issue where the optimiser is doing strange things with filenames a `.`  (period) character. For example:

```js
requirejs.config({paths: {
  'jquery.plugin': 'vendor/jquery.plugin',
  'backbone.plugin': 'vendor/backbone.plugin',
  'backbone.plugin-shim': 'vendor/backbone.plugin-shim',
}});
```

```
src/vendor/jquery.plugin.js -> build/jquery/plugin.js
src/vendor/backbone.plugin.js -> build/backbone/plugin.js 
src/vendor/backbone.plugin-shim.js -> build/backbone.plugin-shim.js
```

This is causing issues with the `onModuleBundleComplete` hook I'm using to generate extra bundle configs, and where I'm expecting `vendor/jquery.plugin.js`, I'm getting  `vendor/jquery/plugin.js`

**Comment by jrburke** [comment]
I am not sure I follow the issue: if the paths are mangled like that in the optimizer, I would not expect it to complete, to reach onModuleBundleComplete. Or is the issue that the r.js optimizer completes, but when inspecting the data passed to onModuleBundleComplete, it has this path issue?

**Comment by jacobbuck** [comment]
I've isolated the issue, it's actually to do with paths that a relative to the baseUrl.

I've set up an example repo here: https://github.com/jacobbuck/rjs789

**Comment by akiellor** [comment]
Also have the same problem.

**Comment by akiellor** [comment]
Created #796 to fix this issue.

**Comment by jacobbuck** [comment]
@akiellor :+1: That pull-request looks good

**Comment by jrburke** [comment]
Fixed by #796.
